core/src/test/scala/org/apache/spark/rpc/RpcEnvSuite.scala

```diff
@@ -162,9 +163,15 @@ abstract class RpcEnvSuite extends SparkFunSuite with BeforeAndAfterAll {
     val rpcEndpointRef = anotherEnv.setupEndpointRef("local", env.address, "ask-timeout")
     try {
       val e = intercept[Exception] {
-        rpcEndpointRef.askWithRetry[String]("hello", 1 millis)
+        rpcEndpointRef.askWithRetry[String]("hello", new RpcTimeout(1 millis, shortProp))
       }
       assert(e.isInstanceOf[TimeoutException] || e.getCause.isInstanceOf[TimeoutException])
+      e match {
+        case te: TimeoutException =>
+          assert(te.getMessage().contains(shortProp))
+        case e: Exception =>
+          assert(e.getCause().getMessage().contains(shortProp))
```

Comment:
ah, I see, that makes sense.  in that case, I kinda prefer we just `intercept[SparkException]` in the first place, and then check the cause & the cause's msg.  And maybe add a brief comment that `askWithRetry` always wraps with `SparkException` so we look at the cause to see the original exception.